specs/interop/optimism-portal-interop.md

```diff
@@ -26,52 +33,86 @@ the corresponding chain would need to deploy and manage its own `SharedLockbox`
+SuperchainConfig when adding this chain to the dependency set.
 
-```mermaid
-sequenceDiagram
-    participant User
-    participant OptimismPortal
-    participant SharedLockbox
+```solidity
+function migrateLiquidity() external;
+```
+
+- MUST only be callable by the `SuperchainConfig` contract
+- MUST set the migrated flag to true
+- MUST transfer all ETH balance to the `SharedLockbox`
+- MUST emit an `ETHMigrated` event with the amount transferred
+
+### Internal ETH Functions
+
+The contract overrides two internal functions from `OptimismPortal2` to handle ETH management with the `SharedLockbox`:
 
-    User->>OptimismPortal: depositTransaction(...)
-    OptimismPortal->>SharedLockbox: lockETH()
-    OptimismPortal->>OptimismPortal: emit TransactionDeposited()
+#### `_lockETH`
+
+Called during deposit transactions to handle ETH locking.
+
+```solidity
+function _lockETH() internal virtual override;
 ```
 
-#### `finalizeWithdrawalTransactionExternalProof`
+- MUST be called during `depositTransaction` when there is ETH value
+- If not migrated, function is a no-op
+- If migrated:
+  - MUST lock any ETH value in the `SharedLockbox`
+  - MUST NOT revert on zero value
 
-Calls `unlockETH` on the `SharedLockbox` with the `tx.value`.
+#### `_unlockETH`
 
-- The function MUST call `unlockETH` on the `SharedLockbox` if:
-  - The token is `ETHER`.
-  - `tx.value` is greater than zero.
-- The ETH is received by the `OptimismPortal` and then sent with the withdrawal transaction
+Called during withdrawal finalization to handle ETH unlocking.
+
+```solidity
+function _unlockETH(Types.WithdrawalTransaction memory _tx) internal virtual override;
+```
 
-```mermaid
-sequenceDiagram
-    participant User
-    participant OptimismPortal
-    participant SharedLockbox
+- MUST be called during withdrawal finalization when there is ETH value
+- If not migrated, function is a no-op
+- If migrated:
+  - MUST unlock the withdrawal value from the `SharedLockbox`
+  - MUST NOT revert on zero value
+  - MUST revert if withdrawal target is the `SharedLockbox`
 
-    User->>OptimismPortal: finalizeWithdrawalTransactionExternalProof(...)
-    OptimismPortal->>SharedLockbox: unlockETH(uint256 value)
-    SharedLockbox->>OptimismPortal: donateETH()
-    OptimismPortal->>OptimismPortal: emit WithdrawalFinalized()
+## Events
+
+### `ETHMigrated`
+
+MUST be triggered when the ETH liquidity is migrated to the SharedLockbox.
+
+```solidity
+event ETHMigrated(uint256 amount);
 ```
 
-### Invariants
+## Invariants
+
+- When processing deposits with ETH value:
+
+  - If migrated, MUST lock the ETH in the `SharedLockbox`
+  - If not migrated, MUST keep the ETH in the portal
+  - The contract MUST NOT revert on zero ETH value operations
+
+- When processing withdrawals with ETH value:
+
+  - If migrated, MUST unlock the ETH from the `SharedLockbox`
+  - If not migrated, MUST use the portal's own ETH balance
```

Comment:
this is a bit tricky because even when migrated, the eth sent to the user is from the portal since the lockbox transfer's  the tx.value to the portal and not to the user. This phrase may make people think that lockbox sends funds directly to the user on `unlockETH`